### PR TITLE
fix: accept and use scope for generating recipe

### LINKED ISSUE #402: bug: meteor "new recipe" command is not using latest recipe requirement
**Describe the bug**
Current `meteor new` command does not use the latest recipe requirements.

generated-recipe.yaml
```
name: sample
version: v1beta1
source:
  type: kafka
  config:     
    broker: "my-broker:6668"
sinks:
  - name: http
    config:       
      # The url (hostname and route) of the http service
      url: http://localhost:3333
      method: POST
      # Additional HTTP headers, multiple headers value are separated by a comma
      headers:
        X-Other-Header: value1, value2
```

running this get below error
```
ERRO[0000] error running recipe                          duration_ms=0 err="failed to setup extractor: could not initiate extractor \"kafka\": urn scope is required to generate unique urn" recipe=sample records_count=0
ERRO[0000] failed to setup extractor: could not initiate extractor "kafka": urn scope is required to generate unique urn
```

The error is because since version `0.5.0`, meteor introduce new required `source.scope` field on recipe.
This field is needed to make sure asset's `urn` is unique by using `scope` as some sort of namespacing feature.

**To Reproduce**
1. Run `meteor new recipe sample -e kafka -s http > recipe.yaml`
2. Copy printed out recipe to a file (`recipe.yaml`)
3. Run `meteor run recipe.yaml`

**Expected behavior**

### Update `new` command
`meteor new recipe` should include `scope` on the generated recipe. It could also make `scope` as part of required input when running the command. e.g. `--scope=my-scope`.

### Docs updated
Somewhere in the docs, it would be helpful if the new `scope` field is being mentioned so that users are not stuck.

### PR DESCRIPTION
**Pull request by sudo-suhas**
Add new mandatory flag --scope to 'meteor new recipe' command for
creating a sample recipe and use the provided scope for generating the
recipe.

BREAKING CHANGE: Method signature for generator.Recipe() changed.

Closes #402.

### CODE REVIEW

File: generator/recipe.go [hunk 1]
```diff
@@ -15,9 +15,13 @@ var file embed.FS
 
 // Template represents the template for generating a recipe.
 type Template struct {
```
Comment (sudo-suhas) [inline generator/recipe.go:17]: any concern with making this struct private? technically change to this struct is also breaking change..

File: generator/recipe.go [hunk 2]
```diff
@@ -15,9 +15,13 @@ var file embed.FS
...
-	Name       string
-	Version    string
-	Source     map[string]string
+	Name    string
+	Version string
+	Source  struct {
+		Type         string
```
Comment (StewartJingga) [inline generator/recipe.go:21]: We can rename `Type` to `Name` since we want to move away from `Type`. wdyt?

File: cmd/new.go [hunk 3]
```diff
@@ -88,14 +89,25 @@ func NewRecipeCmd() *cobra.Command {
...
+				Source:     extractor,
+				Scope:      scope,
+				Sinks:      sinkList,
+				Processors: procList,
+			})
 		},
 	}
 
 	cmd.Flags().StringVarP(&extractor, "extractor", "e", "", "Type of extractor")
+	cmd.Flags().StringVarP(&scope, "scope", "n", "", "URN namespace")
```
Comment (StewartJingga) [inline cmd/new.go:103]: `scope` might not always be URN at the moment.

**Review by StewartJingga — CHANGES_REQUESTED** [review]

File: generator/recipe.go (hunk 1 above)
Comment (StewartJingga) [reply generator/recipe.go:17]: no concern atm, since it is only being used inside the package, we can make it as private.

File: cmd/new.go (hunk 3 above)
Comment (sudo-suhas) [reply cmd/new.go:103]: could you suggest an appropriate description?

File: generator/recipe.go (hunk 2 above)
Comment (sudo-suhas) [reply generator/recipe.go:21]: sure, will do

File: cmd/new.go (hunk 3 above)
Comment (StewartJingga) [reply cmd/new.go:103]: Oh actually I misunderstood what you mean by `URN namespace`, my bad.

We can use `URN's namespace` or keeping it as it is also works for me. wdyt?

Comment (sudo-suhas) [reply cmd/new.go:103]: cool, works

**Review by StewartJingga — APPROVED** [review]

File: generator/recipe.yaml [hunk 4]
```diff
@@ -1,9 +1,10 @@
 name: {{.Name}}
 version: {{.Version}}
 source:
-{{- range $key, $value := .Source }}
-  type: {{$key}}
-  config: {{$value | indent 4}}
+{{- with .Source }}
+  type: {{.Name}}
```
Comment (StewartJingga) [inline generator/recipe.yaml:5]: oh here is as well, we can change `type` to `name`. 🙏

**Review by StewartJingga — CHANGES_REQUESTED** [review]

File: generator/recipe.yaml (hunk 4 above)
Comment (sudo-suhas) [reply generator/recipe.yaml]: Done, apologies for missing it earlier.

**Review by StewartJingga — APPROVED** [review]

**Pull request merged**